tmt/steps/execute/upgrade.py

```diff
@@ -216,12 +250,12 @@ def _perform_upgrade(self, guest):
                 # Create a fake discover from the data in the upgrade path
                 plan = self._get_plan(self._discover_upgrade.testdir)
                 data = self._prepare_remote_discover_data(plan)
-                # Force name
-                data['name'] = 'upgrade-discover-remote'
+                # TODO: upgrade-discover-remote uses **testdir** from previous discover plugin??
+                # Leads to new discover instance look into wrong metadata path.
                 # Override the path so that the correct tree is copied
-                data['path'] = self._discover_upgrade.testdir
+                # data['path'] = self._discover_upgrade.testdir
```

Comment:
This was causing trouble, IIUIC, `path` is a path to metadata root directory, but it already exists at this point, thanks to `self._fetch_upgrade_tasks()` call, and `self._run_discover_upgrade()` will fail when attempting to clone into an existing directory.

Reply:
@FrNecas I think I got it, unsetting `url` did the trick. Thanks for the pointers, `path` is back in play, and used.